maze_transformer/evaluation/eval_model.py

```diff
@@ -85,89 +94,76 @@ def load_model_with_configs(
     return model, config_holder
 
 
-def predict_maze_path(
-    tokens: list[str],
+def predict_maze_paths(
+    tokens_batch: list[list[str]],
     data_cfg: MazeDatasetConfig,
     model: HookedTransformer,
-    include_start_coord: bool = True,
-    n_tokens_pred: int = 8,
+    # Note: The start coord is included in the model input, so max_new_tokens is how many tokens can be predicted AFTER the start. This function returns the full paths, including start coord, so in general the max returned path length is max_new_tokens + 1
+    max_new_tokens: int = 8,
     verbose: bool = False,
-) -> tuple[LatticeMaze, MazePath, MazePath]:
-    """given tokens from a dataset, predict the next tokens with the model, decode both true and predicted to paths
-
-    ### Parameters:
-     - `tokens : list[str]`
-       raw tokens from dataset, containing both maze and true path
-     - `data_cfg : MazeDatasetConfig`
-       config for the dataset
-     - `model : HookedTransformer`
-       model to use for prediction
-     - `n_tokens_pred : int`
-       number of tokens to predict
-     - `**generate_kwargs`
-       additional keyword arguments to pass to `model.generate`
-
-    ### Returns:
-     - `tuple[MazePath, MazePath]`
-       ( path_true, path_predicted )
-    """
-
-    # split the tokens into maze (prompt) and path
-    path_start_token: str = SPECIAL_TOKENS["path_start"]
-    path_start_index: int = tokens.index(path_start_token) + 1
-    maze_tokens: list[str] = tokens[:path_start_index]
-    path_true_tokens: list[str] = tokens[path_start_index:]
-
-    if include_start_coord:
-        # add the first coordinate to `maze_tokens`
-        maze_tokens.append(path_true_tokens[0])
-
-    # encode + pad the maze tokens
-    maze_arr_nopad = model.to_tokens(" ".join(maze_tokens), prepend_bos=False)
+) -> list[list[tuple[int, int]]]:
+    indices_batch: list[ATensor] = []
+    for tokens in tokens_batch:
```

Comment:
Not using wrapped tokenizer? (model.to_tokens(batch))

Reply:
I might be misunderstanding how `to_tokens` works, but it doesn't seem like it would be useful here. It converts a string to a list of tokens, but `tokens` is already a list of tokens because that's what `dataset.mazes_tokens` returns.